src/frontend/src/optimizer/plan_node/stream_watermark_filter.rs

```diff
@@ -59,11 +62,23 @@ impl StreamWatermarkFilter {
 
 impl fmt::Display for StreamWatermarkFilter {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            // TODO(yuhao): display watermark filter expr
-            "StreamWatermarkFilter",
-        )
+        let mut builder = f.debug_struct("StreamWatermarkFilter");
+        let input_schema = self.input.schema();
+
+        let display_watermark_descs: Vec<_> = self
+            .watermark_descs
+            .iter()
+            .map(|desc| {
+                Ok::<_, RwError>(DisplayWatermarkDesc {
+                    watermark_idx: desc.watermark_idx,
+                    expr: ExprImpl::from_expr_proto(desc.get_expr()?)?,
```

Comment:
We have the `ExprImpl` representation when binding the source watermark. Why don't we add a in-memory `WatermarkDesc` representation and avoid transforming from/to the protobuf twice? 🤔

Reply:
The watermark info goes to the meta when binding, and return to the frontend via notify service, which is why there is a protobuf watermarkdesc in the plan node.

Reply:
Get it. So I guess it'll be more intuitive if we introduce an in-memory `WatermarkDesc` in (in-memory) `SourceCatalog` and put the conversion logic there?

Reply:
in that case, each time we create a mv on the source, we need to convert the in memory `WatermarkDesc` (especially the expr) back to a proto desc again to generate a plan.

Reply:
Both ways LGTM. cc @st1page

Reply:
LGTM here because we have used prost everywhere
And because the meta does not have logic to modify that expression.I think the `ExprImpl::from_expr_proto` will not throw an error anyway. 
it should be here but it isn't and LGTM... :hot_face: :crying_cat_face: :sob: